OmgvaPOS/src/DiscountManagement/Repository/DiscountRepository.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.EntityFrameworkCore;
+using OmgvaPOS.Database.Context;
+using OmgvaPOS.DiscountManagement.Enums;
+using OmgvaPOS.DiscountManagement.Models;
+using OmgvaPOS.TaxManagement.Models;
+using OmgvaPOS.TaxManagement.Repository;
+
+namespace OmgvaPOS.src.DiscountManagement.Repository;
+
+public class DiscountRepository : IDiscountRepository
+{
+    private readonly OmgvaDbContext _context;
+    private readonly ILogger<DiscountRepository> _logger;
+
+    public DiscountRepository(OmgvaDbContext context, ILogger<DiscountRepository> logger) {
+        _context = context;
+        _logger = logger;
+    }
+    public Discount AddDiscount(Discount discount) {
+        try {
+            _context.Discounts.Add(discount);
+            _context.SaveChanges();
+            return discount;
+        }
+        catch (Exception ex) {
+            _logger.LogError(ex, "An error occurred while saving discount.");
+            throw new ApplicationException("Error saving discount.");
+        }
+    }
+
+    public List<Discount> GetGlobalDiscounts() {
+        try {
+            return [.. _context.Discounts
+                .Where(d => d.Type == DiscountType.Item)];
```

Comment:
Add filter to filter out archived discounts

Reply:
outdated